selftests/unit/test_job.py

```diff
@@ -93,9 +96,11 @@ def post_tests(self):
                                   logdir=self.tmpdir)
         myjob = JobLogPost(args)
         myjob.create_test_suite()
-        myjob.pre_tests()
-        myjob.run_tests()
-        myjob.post_tests()
+        try:
+            myjob.pre_tests()
+            myjob.run_tests()
+        finally:
+            myjob.post_tests()
```

Comment:
This is not about giving examples. This unittest does not use the stock workflow and it does it incorrectly. So this is only a bugfix, not a demonstration nor suggestion for improvement.